### PR TITLE
feat(docs): Update docs for query and write APIs, making them StreamType specific

### PR DESCRIPTION
**Pull request by stbrody**

### CODE REVIEW

File: docs/streamtypes/caip-10-link/api.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+**Example**
+
+In this example we create a Caip10Link for the ethereum mainnet account *0x0544dcf4fce959c6c4f3b7530190cb5e1bd67cb8* and then associate it with the DID *did:3:k2t6wyfsu4pg0t2n4j8ms3s33xsgqjhtto04mvq8w5a2v5xo48idyz38l7ydki*
+
+```javascript
+import { Caip10Link } from '@ceramicnetwork/stream-caip10-link'
+import { EthereumAuthProvider } from '@ceramicnetwork/blockchain-utils-linking'
+
+const ethProvider = // [...] An ethereum provider for communicating with the ethereum blockchain
```
Comment (oed) [inline docs/streamtypes/caip-10-link/api.md:89]: I would just use `window.ethereum` as an example here as that's what most devs will be familiar with.

File: docs/streamtypes/tile-document/api.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+| `syncTimeoutSeconds` | optional    | number            | How long to wait to hear about the current state of the document from the network | 3 for deterministic documents, 0 otherwise |
+
+[:octicons-file-code-16: API reference](https://developers.ceramic.network/reference/typescript/interfaces/_ceramicnetwork_common.createopts-1.html){:target="_blank"}
+
+
+## Update a document
+Use the [`doc.update()`](https://developers.ceramic.network/reference/typescript/classes/_ceramicnetwork_stream_tile.tiledocument-1.html#update){:target="_blank"} method to update the `content` or `metadata` of an existing TileDocument.
+
+```javascript
+const doc = await TileDocument.create(ceramic, content, metadata, opts)
```
Comment (oed) [inline docs/streamtypes/tile-document/api.md:83]: I would do this instead to make it clear that you can updating an existing document:
```suggestion
const doc = await ceramic.loadStream<TileDocument>(docid)
```

File: docs/streamtypes/tile-document/api.md [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+[:octicons-file-code-16: API reference](https://developers.ceramic.network/reference/typescript/interfaces/_ceramicnetwork_common.loadopts-1.html){:target="_blank"}
+
+## Query a deterministic document
+In the [create](#create-a-document) section, we discussed how to create a document with the `deterministic` parameter set to true. Since this parameter allows us to generate the exact same StreamID if we create two documents with the same `content` and `metadata`, it is possible to "query" the document using the same [`TileDocument.create`](https://developers.ceramic.network/reference/typescript/classes/_ceramicnetwork_stream_tile.tiledocument-1.html#create){:target="_blank"} method that we used to initially create it, without needing to know the StreamID before performing the query. Note we are setting the `CreateOpts` parameters (`anchor` and `publish`) to false so that we are only loading the document and not publishing any changes to the network.
+
+``` javascript
+const doc = TileDocument.create(ceramic,
+  { data: 'this data along with the controller and family uniquely identify this document!' },
```
Comment (oed) [inline docs/streamtypes/tile-document/api.md:165]: This is not right. Deterministic documents needs to set content to `null` otherwise there will be a signature and only the controller of the document can load it deterministically.

File: mkdocs.yml [hunk 4]
```diff
@@ -53,16 +53,18 @@ nav:
     #- Ceramic node: run/ceramic-node.md
     #- Anchor service: run/anchor-service.md
   - StreamTypes:
-      #- Overview:
+      - StreamTypes: streamtypes/streamtypes.md
```
Comment (oed) [inline mkdocs.yml:56]: Seems weird to have *StreamTypes* as a subsection of itself. Why was this changed from *Overview*?

**Review by oed — CHANGES_REQUESTED** [review]

File: docs/streamtypes/tile-document/api.md (hunk 3 above)
Comment (stbrody) [reply docs/streamtypes/tile-document/api.md:165]: Woah, I didn't realize that.  That really restricts how deterministic documents can be used.  That also means that what @PaulLeCam and I came up with for AppendCollection won't actually work: https://github.com/ceramicnetwork/CIP/blob/9ed1a48026917596f59266fba39e79454b55bf16/CIPs/CIP-85/CIP-85.md#deterministic-slice-access

Comment (oed) [reply docs/streamtypes/tile-document/api.md:165]: Note that you can still populate the metadata properties.
This is a security critical restriction.

**Comment by oed** [comment]
Just remembered that we need to update the HTTP API docs as well: https://developers.ceramic.network/reference/http-api/

File: docs/streamtypes/streamtypes.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# StreamTypes
+
+Each Stream in Ceramic is of a specific type. Different StreamTypes can contain different structures of data, and have different APIs for creating, updating, and inspecting them. The currently available StreamTypes are:
+
+* [TileDocument](tile-document/overview.md)
```
Comment (stbrody) [inline docs/streamtypes/streamtypes.md:5]: Add high-level descriptions here

File: docs/streamtypes/caip-10-link/api.md [hunk 6]
```diff
@@ -0,0 +1,126 @@
+# Caip10Link API
+
+This guide demonstrates how to create, update, and query Caip10Links on the Ceramic network using the [HTTP](../../../reference/javascript/clients) and [core](../../../reference/javascript/clients) clients.
+
+## Prerequisites
+You need an [installed client](../../build/installation.md) and an [authenticated user](../../build/authentication.md) to perform writes to Caip10Links on the network during runtime. If you only wish to query existing Caip10Links then you still need an installed client but it doesn't need to be authenticated.
```
[the remark this answers is not in the document]
Comment (stbrody) [reply docs/streamtypes/caip-10-link/api.md:6]: You don't?  Even to set the DID, there's no signature from the DID involved?  Does that mean that the owner of a blockchain address could create a Caip10Link linking the blockchain address to a DID they don't actually control?

File: docs/streamtypes/tile-document/api.md (hunk 3 above)
Comment (stbrody) [reply docs/streamtypes/tile-document/api.md:165]: Updated

File: mkdocs.yml (hunk 4 above)
Comment (stbrody) [reply mkdocs.yml:56]: This is a new section so nothing was changed, but updated to use Overview instead

**Review by stbrody — COMMENTED** [review]
> Just remembered that we need to update the HTTP API docs as well: [developers.ceramic.network/reference/http-api](https://developers.ceramic.network/reference/http-api/)

Will do, in a follow-up PR

File: docs/streamtypes/caip-10-link/api.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (oed) [reply docs/streamtypes/caip-10-link/api.md:6]: Yes that's how it works currently. If you want to verify both ways you should make sure that the caip10-link is listed in the *cryptoAccounts* IDX definition.

**Review by oed — APPROVED** [review]
lgtm. I guess we will merge this after ceramic v1 is out.

File: docs/streamtypes/caip-10-link/api.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (stbrody) [reply docs/streamtypes/caip-10-link/api.md:6]: In theory it would be possible to change Caip10Links to additionally sign the link proof with the linked DID when setting the DID link, right?  Then while only the blockchain->DID link would be *discoverable* without IDX, both the blockchain->DID AND DID->blockchain link would be *verifiable* from just the Caip10Link.  Users who wanted to prove bidirectional linkage could still manually pass around the StreamID without using IDX

[the remark this answers is not in the document]
Comment (stbrody) [reply docs/streamtypes/caip-10-link/api.md]: I filed https://github.com/ceramicnetwork/js-ceramic/issues/1354 to follow-up on this idea. In the meantime I updated the docs here to remove the requirement of an authenticated client to write Caip10Links.

[the remark this answers is not in the document]
Comment (oed) [reply docs/streamtypes/caip-10-link/api.md]: It's possible to implement. Not sure which UX considerations are needed for that.
